File: Data Structures & Algorithms/frequency-of-the-most-frequent-element/submission-4.rs
impl Solution {
    pub fn max_frequency(mut nums: Vec<i32>, k: i32) -> i32 {
        nums.sort();
        let mut l = 0;
        let mut total = 0;
        let mut length = 0;
        for r in 0..nums.len() {
            total += nums[r];
            if ((r - l + 1) as i32) * nums[r] > total + k {
                total -= nums[l];
                l += 1;
            }
            length = max((r - l + 1) as i32, length);
        }
        length
    }
}

File: Data Structures & Algorithms/longest-increasing-subsequence/submission-4.py
class Solution:
    def lengthOfLIS(self, nums: List[int]) -> int:
        dp = [1] * len(nums)
        longest = 0

        for i in range(len(nums)):
            for j in range(0, i):
                if nums[j] < nums[i]:
                    dp[i] = max(dp[i], dp[j] + 1)
            
            longest = max(longest, dp[i])
        
        return longest